crates/turborepo-lib/src/run/mod.rs

```diff
@@ -137,7 +137,12 @@ impl Run {
     }
 
     pub fn create_run_for_non_interruptible_tasks(&self) -> Self {
-        let mut new_run = self.clone();
+        let mut new_run = Self {
+            // ProcessManager is a singleton, so we want to explicitly recreate it
```

Comment:
Non blocking, but I think singleton is the wrong term as that means that only one of these should be created per thread. I think just saying the cloning a ProcessManager shares state is clear enough.